tests/framework/extensions/nodes/ec2/ec2.go

```diff
@@ -149,3 +179,203 @@ func getSSHKeyName(sshKeyName string) string {
 	stringSlice := strings.Split(sshKeyName, ".")
 	return stringSlice[0]
 }
+
+// create PEM for Windows Instances
+func generatePEMKey(client *rancher.Client) (string, error) {
+	ec2Client, err := client.GetEC2Client()
+	if err != nil {
+		return "", err
+	}
+	input := ec2.CreateKeyPairInput{KeyName: aws.String(AppendRandomString(AutomationPemKeyName))}
+	newKey, err := ec2Client.SVC.CreateKeyPair(&input)
+	if err != nil {
+		return "", err
+	}
+	sensitivePEM := newKey.KeyMaterial
+	user, err := user.Current()
+	if err != nil {
+		return "", err
+	}
+	err = os.MkdirAll(filepath.Join(user.HomeDir, sshPath), 0744)
+	if err != nil {
+		return "", err
+	}
+	localPEM := filepath.Join(user.HomeDir, sshPath, aws.StringValue(input.KeyName), ".pem")
+	err = os.WriteFile(localPEM, []byte(convert.ToString(sensitivePEM)), 0400)
+	if err != nil {
+		os.Remove(localPEM)
+		return "", err
+	}
+
```

Comment:
We should cleanup the PEMKey creation, similar to how we delete the nodes using the session cleanup.

Reply:
I added a call to the cleanupPEM function in this err block